src/dsymbol/conversion/package.d

```diff
@@ -145,9 +147,16 @@ class SimpleParser : Parser
 
 	override MissingFunctionBody parseMissingFunctionBody()
 	{
-		skipContracts();
-		if (moreTokens && (currentIs(tok!"do") || current.text == "body"))
+		const bool needDo = skipContracts();
+		if (!needDo)
+		{
+			if (currentIs(tok!"{"))
```

Comment:
main problem was this and caused a mess with the scope because a missing contract was always created.